frontend/src/views/DashboardPage/components/MoveSecrets/DirectorySelector.tsx

```diff
@@ -0,0 +1,143 @@
+        if (!folders) return null;
+
+        const indentation = depth * 20;
+
+        return (
+            <ul style={{ marginLeft: `${indentation}px` }}>
+                {folders.map((folder) => (
+                    <li key={Math.floor(Math.random() * 50000) + 1}>
+                        <button
+                            className={twMerge(`py-1 px-2 bg-mineshaft-500  hover:bg-mineshaft-600 mb-2 rounded-md ${selectedFolder === folder.id ? "bg-mineshaft-400 hover:bg-mineshaft-400" : ""}`)}
+                            onClick={() => handlePathSelection(`${currentPath}/${folder.name}`, folder.id)}
+                            type="submit"
+                        >
+                            {folder.name}
+                        </button>
+                        {folder.folders && renderRootDirectory(folder.folders, `${currentPath}/${folder.name}`, depth + 1)}
+                    </li>
+                ))}
+            </ul>
+        );
+    };
+
+    const renderCurrentDirectory = (folders: Folder[] | undefined, currentPath: string) => {
+        if (!folders) return null;
+
+        return (
+            <ul className="mt-3">
+                {folders.map((folder) => (
+                    <li key={folder.id}>
+                        <button
+                            className="py-1 px-2 bg-mineshaft-500 mb-2 hover:bg-mineshaft-600 text-gray-400 rounded-md"
+                            onClick={() => handlePathSelection(`${currentPath}/${folder.name}`, folder.id)}
+                            type="submit"
+                        >
+                            {folder.name}
+                        </button>
+                    </li>
+                ))}
+            </ul>
+        );
+    };
+
+    const adjustedPath = selectedPath.replace(/\/root/g, "/").replace(/\/{2,}/g, "/");
+
+
+    return (
+        <div className="p-4 border border-mineshaft-600 rounded-lg shadow-md">
+            {
+                (directoryTree.length > 0) && (
+                    <div>
+                        <h2 className="text-md font-semibold  text-gray-300">Select a folder path within the root directory:</h2>
+                        {renderRootDirectory(directoryTree, "")}
+                    </div>
+                )
+            }
+
+            {
+                (directoryData?.folders && directoryData?.folders.length > 0) && (
+                    <div>
+                        <h2 className="text-md font-semibold text-gray-300">Select a folder within current directory:</h2>
+                        {renderCurrentDirectory(directoryData?.folders, "")}
+                    </div>
+                )
+            }
+            <p className="text-sm mt-2 text-gray-300 mt-5">Selected folder path: <span className="rounded-md bg-mineshaft-600 py-1 px-1.5 ml-2">{adjustedPath}</span></p>
+        </div>
+    );
+};
+
+export default DirectorySelector;
```

Comment:
We prefer named export rather than default ones

Reply:
Addressed!